### PR TITLE
feat: `orb sphere history` and `orb sphere render`

### PR DESCRIPTION
**Pull request by cdata**
Also fixes gateway DOS bug resulting from impossible-to-resolve link records.

`orb sphere render` forces a re-render of the user-visible filesystem content in a workspace. This can include changing the render depth of the graph; if the render depth is decreased from it's previous value, this will result in the removal of peers that were previously visible from the visible file system hierarchy.

`orb sphere history` shows the change history of the sphere. You can see an example of this here:

```
❯ orb sphere history
Version: bafyr4ic77uiynexgdsj3shmzkjbzwbupgzwggs6bmmrcccfbuol654rgam (Latest)
Author: did:key:z6MkmwXmvVmS1ZCLwrdpTEM273PjPWtpnGreE7HnkrMpfhHG
No content, address book or authority changes

Version: bafyr4ib5ctwgsugaqszor3z3cl6mxsfcropf3zfgv7xcoxlgia7opbagsi
Author: did:key:z6Mkrz9JWbSP1D5CzBEfq5psVh4JDPaQ5qvnVz4vaSwvSe7U
Changes:

     Updated  peer2-renamed -> bafkr4idtbp22w4xdjeo3k6co27zrmozicufeteaw6thu474tzoydqhwyoq

Version: bafyr4ietttfmdd6trf3wmvjsvvobljqpkcruytnflkv5zsatu5k2jdamau
Author: did:key:z6Mkrz9JWbSP1D5CzBEfq5psVh4JDPaQ5qvnVz4vaSwvSe7U
Changes:

    Followed  peer2-renamed -> did:key:z6Mkr3is8GGvfqMF1GhY9STi4VGu4e2i3BPSRihtrrN12dr5

  Unfollowed  peer3
  Unfollowed  peer2

Version: bafyr4igqnyjbqrpphnpklr2dkqned4tlz52qzkjeoyld2ag37ucykwwwya
Author: did:key:z6Mkrz9JWbSP1D5CzBEfq5psVh4JDPaQ5qvnVz4vaSwvSe7U
Changes:

   Delegated  cli (did:key:z6MkmwXmvVmS1ZCLwrdpTEM273PjPWtpnGreE7HnkrMpfhHG)

Version: bafyr4idojhwato5pyubdaxl4vmvoyjej67ft4676buh5s53biyrihwgwze
Author: did:key:z6Mkrz9JWbSP1D5CzBEfq5psVh4JDPaQ5qvnVz4vaSwvSe7U
Changes:

     Updated  peer2 -> bafkr4ibwgb3pji2p4dgt7ook7cq27dugyw63rb5s5qx35wnrvtkbcubqy4
     Updated  peer3 -> bafkr4ifv37a2ngppkes6ztifnzhmfn2hbuneswjgm5z4e5o2rc5wp2rlmm

Version: bafyr4ihwf5p4q35tmpibze2gta43h72yet5e36jjyzlza22iogs7q4wguq
Author: did:key:z6Mkrz9JWbSP1D5CzBEfq5psVh4JDPaQ5qvnVz4vaSwvSe7U
Changes:

    Followed  peer2 -> did:key:z6Mkr3is8GGvfqMF1GhY9STi4VGu4e2i3BPSRihtrrN12dr5
    Followed  peer3 -> did:key:z6MkvLYSwSL68reYoejH68Vga14BVUzhaiGWFrBKHNsNtjpA

Version: bafyr4ievnlpxhlfn3ruozytnzjtvn3kjdiyobtcx3x7hz3jhzq2lwpeyju
Author: did:key:z6Mkrz9JWbSP1D5CzBEfq5psVh4JDPaQ5qvnVz4vaSwvSe7U
Changes:

    Modified  /content1

Version: bafyr4ieivlfbwsscwhxreulwmppq35di7whcglpbsa3lsjzsd7cxithfza
Author: did:key:z6MkismXxNXRpmu8uKsXrNojDkb3mwbDwFawSWHV9eu1Toxo
Changes:

   Delegated  (OWNER) (did:key:z6Mkrz9JWbSP1D5CzBEfq5psVh4JDPaQ5qvnVz4vaSwvSe7U)

Version: bafyr4iaf7rdj2jbbxcctw4qw4cply5dj7wjgpz4kwyjvnezc7vwjmphpku
Author: did:key:z6MkismXxNXRpmu8uKsXrNojDkb3mwbDwFawSWHV9eu1Toxo
This is the base version of the sphere
No content, address book or authority changes

Start of history
```

### CODE REVIEW

File: rust/noosphere-core/src/helpers.rs [hunk 1]
```diff
@@ -0,0 +1,48 @@
+// TODO: Figure out how to share this code across crates
```
Comment (jsantell) [inline rust/noosphere-core/src/helpers.rs:1]: Is this still relevant? `noosphere-sphere` seems to be using it after moving to core

File: rust/noosphere-storage/src/retry.rs [hunk 2]
```diff
@@ -22,20 +63,33 @@ pub struct BlockStoreRetry<S>
...
 }
 
-impl<S> BlockStoreRetry<S>
+impl<S> From<S> for BlockStoreRetry<S>
 where
     S: BlockStore,
 {
-    pub fn new(store: S, max_retries: u32, timeout: Duration) -> Self {
-        BlockStoreRetry {
+    fn from(store: S) -> Self {
```
Comment (jsantell) [inline rust/noosphere-storage/src/retry.rs:86]: nice!

**Review by jsantell — APPROVED** [review]
great work!

File: rust/noosphere-core/src/helpers.rs (hunk 1 above)
Comment (cdata) [reply rust/noosphere-core/src/helpers.rs:1]: Ah, good catch. Yah, this is TODONE.

**Pull request merged**